staking/README.md

```diff
@@ -0,0 +1,541 @@
+# Staking Module
+
+## Table of Contents
+
+* [Preliminary terminology](#preliminary-terminology)
+* [Overview](#overview)
+* [How does one join the network as a validator](#how-does-one-join-the-network-as-a-validator)
+* [How to propose a MsgValidatorJoin transaction](#how-to-propose-a-msgvalidatorjoin-transaction)
```

Comment:
Don't you think the `propose` sections are sub-paragraphs of the previous one?
e.g.
```
* [How does one join the network as a validator](#how-does-one-join-the-network-as-a-validator)
** [How to propose a MsgValidatorJoin transaction](#how-to-propose-a-msgvalidatorjoin-transaction)
```

Reply:
Good idea, though I just followed the format from `bor` module.

Reply:
Check how this looks: [159a33b](https://github.com/maticnetwork/heimdall/pull/1082/commits/159a33bafa2f43c7cc2ae0f9d57b4257ef036436)

Reply:
I actually checked and apparently you need a `tab` (or two spaces) and not a double `*` to make it a sub-paragraph, sorry for the confusion (that works for headers like # and ##).
So in this case it'd be
```
* [How does one join the network as a validator](#how-does-one-join-the-network-as-a-validator)
  * [How to propose a MsgValidatorJoin transaction](#how-to-propose-a-msgvalidatorjoin-transaction)
```

Reply:
And now: [0ce7e54](https://github.com/maticnetwork/heimdall/pull/1082/commits/0ce7e545e376b2cc4cb0b4027731f326607523fa) ?